### DISCUSSION TITLE
Problema actividad 4, grupo 2

### DISCUSSION THREAD

**Discussion by ValentinMilocco** [frlp-utn-ingsoft/catedra#145] [Q&A] [answered]
Hola, tengo un problema, modifique el models y el views para las validaciones del telefono pero tengo un problema a la hora de mostrar el error por pantalla, modifique algo el template, pero sigue sin funcionar correctamente, el link de la branch :https://github.com/JuaniLezcano/vetsoft/tree/Feature/phone-starts-with-54
![Sin título](https://github.com/frlp-utn-ingsoft/catedra/assets/132497692/afda3f09-fe4c-4bae-a7dc-22e3f67ae853)

**Comment by RodrigoJacznik** [comment]
Hola Valentin. El problema es que no te marca el input de color rojo?

**Comment by ValentinMilocco** [comment]
claro

**Reply by RodrigoJacznik** [reply]
El problema esta en que las validaciones server-side no tienen que agregarle la clase `was-validated` al elemento `form`. https://github.com/JuaniLezcano/vetsoft/blob/036718b0487a30d1d9daaabea233dce0081d6b5f/app/templates/clients/form.html#L13

Ahora, si le sacas esa clase al form, vas a tener que agregar el codicional que agrega el `is-invalid` a cada input como se hace en este input. 
https://github.com/JuaniLezcano/vetsoft/blob/036718b0487a30d1d9daaabea233dce0081d6b5f/app/templates/clients/form.html#L43

**Reply by ValentinMilocco** [reply]
no me funciona, puse en cada input el is-invalid y borre la clase was-validated al elemento form, pero deja de mostrar los errores en los otros campos:
![Sin título1](https://github.com/frlp-utn-ingsoft/catedra/assets/132497692/2ead6397-149a-4564-bada-310aa7ad1a97)

**Reply by ValentinMilocco** [reply]
En el alta del cliente funciona correctamente pero no en el editar

**Reply by RodrigoJacznik** [reply]
pushealo asi lo puedo revisar.

**Reply by ValentinMilocco** [reply]
ahi lo pushee: https://github.com/JuaniLezcano/vetsoft/tree/Feature/phone-starts-with-54

**Reply by RodrigoJacznik** [reply]
Perdon, te recomende algo que te iba a complicar. (hay que meter un poco de js para que funcione de esa forma).
Creo que una mejor solucion para este caso es agregarle el atributo pattern al campo de telefono.

El formulario te deberia quedar asi

```html
<form class="vstack gap-3 {% if errors %}was-validated{% endif %}"
    aria-label="Formulario de creacion de cliente"
    method="POST"
    action="{% url 'clients_form' %}"
    novalidate>

    {% csrf_token %}

    <input type="hidden" value="{{ client.id }}" name="id" />

    <div>
        <label for="name" class="form-label">Nombre</label>
        <input type="text"
            id="name"
            name="name"
            value="{{client.name}}"
            class="form-control"
            required/>

        {% if errors.name %}
            <div class="invalid-feedback">
                {{ errors.name }}
            </div>
        {% endif %}
    </div>
    <div>
        <label for="phone" class="form-label">Teléfono</label>
        <input type="text"
            id="phone"
            name="phone"
            type="tel"
            class="form-control"
            value="{{ client.phone|default_if_none:'' }}"
            required
            pattern="54\d{11}"/>

        {% if errors.phone %}
            <div class="invalid-feedback">
                {{ errors.phone }}
            </div>
        {% endif %}
    </div>
    <div>
        <label for="email" class="form-label">Email</label>
        <input type="email"
            id="email"
            name="email"
            class="form-control"
            value="{{ client.email }}"
            required/>

        {% if errors.email %}
            <div class="invalid-feedback">
                {{ errors.email }}
            </div>
        {% endif %}
    </div>
    <div>
        <label for="address" class="form-label">Dirección</label>
        <input type="text"
            id="address"
            name="address"
            value="{{client.address}}"
            class="form-control" />

        <div class="invalid-feedback">
            La dirrección es opcional
        </div>
    </div>

    <button class="btn btn-primary">Guardar</button>
</form>
```

**Reply by ValentinMilocco** [reply]
pero con ese codigo no muestra el texto de los demas errores:
![sintitulo2](https://github.com/frlp-utn-ingsoft/catedra/assets/132497692/35fe8675-d34b-4a65-bf15-c07dccb7ce76)

**Reply by RodrigoJacznik** [reply, marked as answer]
Revisalo porque creo que algo te falto. Te mando una captura de como veo tu proyecto con el formulario que te envié.

![image](https://github.com/frlp-utn-ingsoft/catedra/assets/2715743/d2e81b13-c7ae-4ed3-b41a-67f7bf1fee9e)

**Reply by ValentinMilocco** [reply]
ahi pushee lo de ambos tanto veterinario como cliente, yo sigo sin ver los errores al editar cada uno de ellos, quizas es un problema de mi pc

**Reply by RodrigoJacznik** [reply]
Yo veo los mensajes de error en ambos formularios

**Reply by ValentinMilocco** [reply]
Bueno me quedo tranquilo entonces, Muchas gracias Rodri.
